components/layout_2020/display_list/stacking_context.rs

```diff
@@ -420,38 +509,32 @@ impl StackingContext {
         // Let’s find the corresponding fragment.
 
         // The fragment generated by the root element is the first one here, unless…
-        let first_if_any = self.fragments.first().or_else(|| {
+        let first_if_any = self.contents.first().or_else(|| {
             // There wasn’t any `StackingContextFragment` in the root `StackingContext`,
             // because the root element generates a stacking context. Let’s find that one.
-            self.stacking_contexts
+            self.real_stacking_contexts_and_positioned_stacking_containers
                 .first()
                 .and_then(|first_child_stacking_context| {
-                    first_child_stacking_context.fragments.first()
+                    first_child_stacking_context.contents.first()
                 })
         });
 
-        macro_rules! debug_panic {
-            ($msg: expr) => {
-                if cfg!(debug_assertions) {
-                    panic!($msg)
-                }
-            };
-        }
-
         let first_stacking_context_fragment = if let Some(first) = first_if_any {
             first
         } else {
             // This should only happen if the root element has `display: none`
-            debug_panic!("`CanvasBackground::for_root_element` should have returned `style: None`");
-            return;
+            panic!("`CanvasBackground::for_root_element` should have returned `style: None`");
         };
 
-        let fragment = first_stacking_context_fragment.fragment.borrow();
+        let StackingContextContent::Fragment { fragment, scroll_node_id, containing_block, .. }
+            = first_stacking_context_fragment else {
+                panic!("Expected a fragment, not a stacking container");
+            };
+        let fragment = fragment.borrow();
         let box_fragment = match &*fragment {
             Fragment::Box(box_fragment) | Fragment::Float(box_fragment) => box_fragment,
             _ => {
-                debug_panic!("Expected a box-generated fragment");
-                return;
+                panic!("Expected a box-generated fragment");
```

Comment:
Ditto. Failing gracefully is almost certainly better for users here.

Reply:
done in /files/e2df884f1c60ec28793f8a9a2a1f5d37e6dba028#diff-82520938a024f875c789f7314111aaeb3f11a47fd9b956a3a90fef8399ddaea1

Reply:
done in 43692821f9bccde5eda1b5651e21c535bfaebe9d